File: task-api/src/main/scala/pwr/kgl/taskapi/DataModels.scala
package pwr.kgl.taskapi

case class ServiceInformation(val endpoints: List[SingleEndpointConfiguration])

case class SingleEndpointConfiguration(
                                        val name: String,
                                        val path: String,
                                        val method: String,
                                        val input: List[FieldDefinition],
                                        val output: List[FieldDefinition],
                                        consumes: Option[String],
                                        produces: Option[String]
                                      )

case class FieldDefinition(
                            val name: String,
                            val `type`: String,
                            val required: Boolean,
                            val additional_description: Option[String]
                          )

case class ShortInfo(name:String, endpoints:List[String])

File: micro-frontend/src/app/datamodels.ts
export interface ShortInfo{
  name: string;
  endpoints: string[];
}

export interface SingleEndpointConfiguration{
  name: string;
  path: string;
  method: string;
  input: FieldDefinition[];
  output: FieldDefinition[];
  consumes?: string,
  produces?: string
}

export interface FieldDefinition{
  name: string;
  type: string;
  required: boolean;
  additional_description?: string

}

